app/Http/Controllers/Auth/OauthLoginController.php

```diff
@@ -21,23 +21,30 @@ public function __construct()
 
     public function login()
     {
-        return Socialite::driver('mediawiki')
-            ->redirect();
+        try {
+            return Socialite::driver('mediawiki')->redirect();
+        } catch (Exception $e) {
+            //Cathing temporary credentials retrival error
+            //We are not concerned about logging this as it's a user issue, easily fixed.
+            abort(400); 
+        }
     }
 
     public function callback(Request $request)
     {
         // run this inside a transaction.
         // helps mainly for development stuff, when loading permissions fails
         $user = DB::transaction(function () use ($request) {
-            $socialiteUser = Socialite::driver('mediawiki')->user();
 
-            // sanity check
-            if (!$socialiteUser->getId() || !$socialiteUser->getName()) {
-                Log::error('OAuth failed: MediaWiki api returned an invalid user object ' . json_encode($socialiteUser));
-                abort(500);
+            try {
+                $socialiteUser = Socialite::driver('mediawiki')->user();
+            } catch (Exception $e) {
+                //the previous sanity check code didn't account for OAuth tokens being undefinited, and therefore causing an error
+                //We are not concerned about logging this as it's a user issue, easily fixed.
+                abort(400); 
             }
```

Comment:
Not a fan of suppressing all errors, would like to know more to be able to consider alternatives. What kinds of errors is this throwing? https://phabricator.wikimedia.org/T272319 might also be relevant here.